clang/test/Driver/sycl-linker-wrapper.cpp

```diff
@@ -236,7 +236,52 @@
 // CHK-SYCLBIN-CMDS-NEXT: "{{.*}}llvm-link" -only-needed --suppress-warnings [[FIRSTLLVMLINKOUT]].bc {{.*}}.bc -o [[SECONDLLVMLINKOUT:.*]].bc
 // CHK-SYCLBIN-CMDS-NEXT: "{{.*}}sycl-post-link" {{.*}} SYCL_POST_LINK_OPTIONS -o [[SYCLPOSTLINKOUT:.*]].table [[SECONDLLVMLINKOUT]].bc
 // CHK-SYCLBIN-CMDS-NEXT: "{{.*}}llvm-spirv" {{.*}} -o {{.*}}
-// CHK-SYCLBIN-CMDS-NOT:  offload-wrapper: input
+// CHK-SYCLBIN-CMDS-NOT:  offload-wrapper: output
 // CHK-SYCLBIN-CMDS-NOT: "{{.*}}clang"
 // CHK-SYCLBIN-CMDS-NEXT: "{{.*}}cp" {{.*}}.syclbin a.out
 // CHK-SYCLBIN-CMDS-NOT: "{{.*}}ld"
+
+// Check that clang-linker-wrapper errors out in case when it observes different compilation options encoded in images with the same triple and arch.
+// RUN: clang-offload-packager -o %t.packaged_1.fat "--image=file=%t.o,triple=spir64-unknown-unknown,arch=generic,kind=sycl,compile-opts=aaa,link-opts=bbb"
+// RUN: clang-offload-packager -o %t.packaged_2.fat "--image=file=%t.o,triple=spir64-unknown-unknown,arch=generic,kind=sycl,compile-opts=ccc,link-opts=ddd"
+//
+// RUN: %clang -cc1 %s -triple=x86_64-unknown-linux-gnu -emit-obj -o %t.embeded.o -fembed-offload-object=%t.packaged_1.fat -fembed-offload-object=%t.packaged_2.fat
```

Comment:
I have a few questions to better understand the compilation flow for this specific invocation:
Why are we passing `-cc1` here?
My understanding is we embed the output from `clang-offload-packager` during SYCL host compilation `(-fsycl-is-host)` and this results in the fat object (<host object, device code IR> pair )

Also, does this invocation invoke `clang-offload-packager` again?